docs/footprint.md

```diff
@@ -0,0 +1,22 @@
+# Mender MCU memory footprint
+
+The Mender MCU project is designed with memory footprint in mind. This document
+estimates the memory footprint of the project. This is, the amount of code that
+will end up in the ROM section of the memory and the amount of RAM required.
```

Comment:
My conclusion from our attempts to find some precise numbers is that this is a very strong statement. We can publish the numbers provided by the Zephyr's memory reporting tools below, but *How much extra RAM/ROM memory will be needed if I add Mender MCU to my (Zephyr) application?* is a completely different story. So, TBH, I'd drop this document completely. If we really need it then let's make super-sure everybody reading it clearly understands where those numbers come from and what they really mean.